artio-codecs/src/main/java/uk/co/real_logic/artio/dictionary/DictionaryParser.java

```diff
@@ -443,7 +443,10 @@ private void identifyDuplicateFieldDefinitionsForMessages(final List<Message> me
 
         if (errorMessage.length() > 0)
         {
-            throw new IllegalStateException(errorMessage.toString());
+            if (!Boolean.getBoolean(CodecGenerationTool.VALIDATION_ALLOW_DUPLICATES))
+            {
+                throw new IllegalStateException(errorMessage.toString());
```

Comment:
You should also update the error message in order to tell users that this option can be used in order to switch the duplicate field validation off.

Reply:
Done